src/controllers/oracle.controller.ts

```diff
@@ -0,0 +1,265 @@
+import {repository} from '@loopback/repository';
+import {HttpErrors, operation, param, requestBody} from '@loopback/rest';
+import {GoodOracle, GoodOracleInput} from '../models';
+import {GoodOracleRepository} from '../repositories';
+
+export class OracleController {
+  constructor(
+    @repository(GoodOracleRepository)
+    public goodOracleRepository: GoodOracleRepository,
+  ) {}
+
+  /**
+   * Create a new Oracle
+   *
+   * @param id The PoG ID of the oracle
+   * @param oracle
+   */
+  @operation('post', '/oracle', {
+    summary: 'Create an Oracle',
+    operationId: 'post-oracle',
+    responses: {
+      '201': {
+        description: 'Oracle Created',
+      },
+      '400': {
+        description: 'Missing Required Information',
+        content: {
+          'application/json': {
+            schema: {
+              $ref: '#/components/schemas/ErrorResponse',
+            },
+          },
+        },
+      },
+      '401': {
+        description: 'Unauthorized',
+        content: {
+          'application/json': {
+            schema: {
+              $ref: '#/components/schemas/ErrorResponse',
+            },
+          },
+        },
+      },
+      '404': {
+        description: 'Oracle Not Found',
+        content: {
+          'application/json': {
+            schema: {
+              $ref: '#/components/schemas/ErrorResponse',
+            },
+          },
+        },
+      },
+    },
+    requestBody: {
+      content: {
+        'application/json': {
+          schema: {
+            $ref: '#/components/schemas/GoodOracle_Input',
+          },
+          examples: {
+            'Create an Oracle': {
+              value: {
+                name: 'Leyline',
+                goodOracleURI: 'leyline.gg',
+              },
+            },
+          },
+        },
+      },
+      description: '',
+    },
+    description: 'Create a new Oracle',
+    parameters: [],
+    security: [
+      {
+        Oracle_API_Key: [],
+      },
+    ],
+  })
+  async postOracle(
+    @requestBody({
+      content: {
+        'application/json': {
+          schema: {
+            $ref: '#/components/schemas/GoodOracle_Input',
+          },
+          examples: {
+            'Create an Oracle': {
+              value: {
+                name: 'Leyline',
+                goodOracleURI: 'leyline.gg',
+              },
+            },
+          },
+        },
+      },
+      description: '',
+    })
+    oracle: GoodOracle,
+  ): Promise<unknown> {
+    oracle.id = 0;
+    return this.goodOracleRepository.create(oracle);
```

Comment:
```suggestion
    oracle: Omit<GoodOracle, 'id'>,
  ): Promise<unknown> {
    return this.goodOracleRepository.create(oracle);
```
This will take care of removing the id from the type and create a correct resource